lib/linter-liferay.js

```diff
@@ -61,6 +80,28 @@ export default class LinterLiferay {
       'linter-liferay.checkSFPath',
       newVal => this._cmd = newVal
     ))
+
+    this._subscriptions.add(atom.config.observe(
+      'linter-liferay.requireConfig',
+      newVal => this._requireConfig = newVal
+    ))
+
+    this._pathMap = {}
+
+    this._subscriptions.add(
+      atom.workspace.onDidOpen(event => {
+        const {uri} = event
+
+        if (this._requireConfig && !(uri in this._pathMap)) {
+          if (getConfigPath(uri)) {
```

Comment:
Instead of this `if/else`, just assign the result of `getConfigPath`:
```js
this._pathMap[uri] = getConfigPath(uri)
```
Should be fine after we change `getConfigPath` to always return a boolean instead.